packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -328,6 +325,124 @@ const EDiscovery = WebexPlugin.extend({
+    }
+
+    // use spread operator to set default options
+    options = {...this.config.defaultOptions, ...options};
+
+    const [requestOptions, reportUUID] = this._createRequestOptions(reportId, '/contents/container');
+
+    return this.request({
+      method: 'GET',
+      ...requestOptions,
+      qs: {offset: options.offset, size: options.size},
+      timeout: options.timeoutMs
+    })
+      .then(async (res) => {
+        await this._writeToContentContainerCache(reportUUID, res.body);
+
+        return Promise.resolve(res);
+      });
+  },
+
+  @waitForValue('@')
+  @oneFlight({keyFactory: (reportId, options) => createOneFlightKey(reportId, options)})
+  /**
+   * Retrieves information for a specific content container relevant to a specified report
+   * @param {UUID|string} reportId - UUID or url of the report being requested
+   * @param {UUID} containerId - Id of the contenyt container being requested
+   * @param {Object} options - optional parameters for this method
+   * @param {number} options.timeoutMs - connection timeout in milliseconds, defaults to 30s
+   * @returns {Promise<ResponseEntity<ContentContainer>>} Http response containing the specified content container
+   */
+  getContentContainerByContainerId(reportId, containerId, options) {
+    if (!reportId) {
+      throw Error('Undefined parameter');
+    }
+
+    // use spread operator to set default options
+    options = {...this.config.defaultOptions, ...options};
+
+    const [requestOptions, reportUUID] = this._createRequestOptions(reportId, `/contents/container/${containerId}`);
+
+    // If this content container has already been cached then it can be retrieved from there instead of making a network call to ediscovery
+    if (this.contentContainerCache.has(reportUUID) && this.contentContainerCache.get(reportUUID).has(containerId)) {
+      return {body: this.contentContainerCache.get(reportUUID).get(containerId), statusCode: 200};
+    }
+
+    this.logger.warn(`Cache miss for container ${containerId} in contentContainerCache`);
+
+    return this.request({
+      method: 'GET',
+      ...requestOptions,
+      qs: {offset: options.offset, size: options.size},
+      timeout: options.timeoutMs
+    })
+      .then(async (res) => {
+        await this._writeToContentContainerCache(reportUUID, [res.body]);
+
+        return Promise.resolve(res);
+      });
+  },
+
+  /**
+   * The results of a getContentContainer or getContentContainerByContainerId request are written to the contentContainerCache
+   * since information for a container is likely to be accessed very frequently when decrypting activities
+   * @param {UUID} reportId - Id of the report which contains the relevant content summary
+   * @param {Array<Object>} containers - List of the container objects to be written to the cache
+   * @returns {Promise} - Promise resolution indicating operation is complete
+   */
+  _writeToContentContainerCache(reportId, containers) {
```

Comment:
Slightly wish `ContentContainer` had been simplified to `Container`. But that's a matter of taste.